packages/fury-adapter-apib-serializer/lib/filters.js

```diff
@@ -34,18 +34,11 @@ const indent = (input, spaces, options = { first: false }) => {
   * use a shorthand syntax in the template.
   */
 const bodyOnly = (payload) => {
-  let headers;
+  // Headers, excluding content-type as that goes outside payload
+  // (e.g. `+ Response (application/json)`)
+  const hasHeaders = payload.headers && payload.headers.exclude('Content-Type').length > 0;
 
-  // First, we need to filter out the content-type header. This is handled
-  // outside of the payload (e.g. `+ Response (application/json)`)
-  if (payload.headers) {
-    headers = payload.headers.exclude('Content-Type');
-  }
-
-  return payload.messageBody && !(
-    headers.length || payload.dataStructure || payload.messageBodySchema
```

Comment:
This is the particular line, `headers.length` which causes the exception as `headers` is undefined and thus you cannot call `length` on it.